### PR TITLE
Feature/tao 3872 implement verbose logger

### PR DESCRIPTION
**Pull request by siwane**
Requires https://github.com/oat-sa/generis/pull/356

### CODE REVIEW

File: models/classes/cliArgument/argument/implementation/verbose/Debug.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+     * Check if params array contains targeted arguments, Short and Long
+     * In case of Long, check is done is following param argument
+     *
+     * @param array $params
+     * @return bool
+     */
+    public function isApplicable(array $params)
+    {
+        return in_array('-vvv', $params)
+            || (in_array('--verbose', $params) && ($params[array_search('--verbose', $params)+1] == 3));
```
Comment (chilly86) [inline models/classes/cliArgument/argument/implementation/verbose/Debug.php]: undefined offset

File: models/classes/cliArgument/argument/implementation/verbose/Error.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+     * Check if params array contains targeted arguments, Short and Long
+     * In case of Long, check is done is following param argument
+     *
+     * @param array $params
+     * @return bool
+     */
+    public function isApplicable(array $params)
+    {
+        return in_array('-v', $params)
+            || (in_array('--verbose', $params) && ($params[array_search('--verbose', $params)+1] == 1));
```
Comment (chilly86) [inline models/classes/cliArgument/argument/implementation/verbose/Error.php]: undefined offset

File: models/classes/cliArgument/argument/implementation/verbose/Notice.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+     * Check if params array contains targeted arguments, Short and Long
+     * In case of Long, check is done is following param argument
+     *
+     * @param array $params
+     * @return bool
+     */
+    public function isApplicable(array $params)
+    {
+        return in_array('-vv', $params)
+            || (in_array('--verbose', $params) && ($params[array_search('--verbose', $params)+1] == 2));
```
Comment (chilly86) [inline models/classes/cliArgument/argument/implementation/verbose/Notice.php]: undefined offset

**Comment by siwane** [comment]
Comments solved

**Comment by chilly86** [comment]
I'd go with a more simple solution. 

1. A helper/function to decide the verbose level based on argv. (0-3)

2. A factory which is creating a logger instance:
getInstance($verboseLevel)
switch ($verboseLevel)
case: 0
return new NullLogger
case: 1
return new VerboseLogger(LogLevel::ERROR)
....

3. Setting this instance to LoggerService

**Pull request merged**